scripts/wrappers/addons.py

```diff
@@ -41,13 +161,7 @@ def add(name: str, repository: str, reference: str, force: bool):
     subprocess.check_call(cmd)
     subprocess.check_call(["chgrp", "microk8s", "-R", repo_dir])
 
-    if not (repo_dir / "addons.yaml").exists():
-        click.echo(
-            "Error: repository '{}' does not contain an addons.yaml file".format(name), err=True
-        )
-        click.echo("Remove it with:", err=True)
-        click.echo("    microk8s addons repo remove {}".format(name), err=True)
-        sys.exit(1)
+    validate_addons_repo(repo_dir)
```

Comment:
if this fails in any way, i think it is best that we cleanup the folder instead of leaving it there. in general, i'd rather we didn't call sys.exit(1) in multiple place, but rather an exception that we can catch here, print a useful error message and then cleanup and exit.

Reply:
Sounds like a much better idea 👍🏼